rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1798,6 +1798,16 @@ public BooleanWrapper releaseNodes(NodeSet nodes) {
                         if (rmnode.isToRemove()) {
                             removeNodeFromCoreAndSource(rmnode, caller);
                             nodesReleased.add(node);
+                            if (delayedNodeSourceRemovalEvents.containsKey(rmnode.getNodeSourceName()) &&
+                                nodeSourceCanBeRemoved(rmnode.getNodeSourceName())) {
+
+                                final RMNodeSourceEvent removedEvent = delayedNodeSourceRemovalEvents.remove(rmnode.getNodeSourceName());
+
+                                logger.info(NODE_SOURCE_STRING + rmnode.getNodeSourceName() +
+                                            " has been successfully " + removedEvent.getEventType().getDescription());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal " has been successfully " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

Reply:
👍